src/lib/crypto.c

```diff
@@ -518,10 +521,14 @@ pgp_encrypt_buf(rnp_ctx_t *         ctx,
     }
 
     /* Push the encrypted writer */
-    pgp_push_enc_se_ip(output, pubkey, ctx->ealg);
+    if (!pgp_push_enc_se_ip(output, pubkey, ctx->ealg, insize)) {
+        return false;
```

Comment:
pgp_setup_memory_write will allocate ``output``, so here we will leak memory. 
I would suggest set ``output=NULL`` and ``outmem=NULL, have ``goto end`` here and have ``end`` label just before leaving this function (see for example ``ecdsa.c``). We also need to clear outmem in case of error